### PR TITLE
fix(http): convert user errors

### PR DESCRIPTION
**Pull request by kelwang**
Closes https://github.com/influxdata/platform/issues/1300

### CODE REVIEW

File: bolt/basic_auth_service.go [hunk 1]
```diff
@@ -23,8 +23,9 @@ func (c *Client) setPassword(ctx context.Context, tx *bolt.Tx, name string, pass
 		return err
 	}
 
-	u, err := c.findUserByName(ctx, tx, name)
-	if err != nil {
+	u, pe := c.findUserByName(ctx, tx, name)
+	if pe != nil {
```
[the remark this answers is not in the document]
Comment (kelwang) [reply bolt/basic_auth_service.go]: Oh, it's an go ugly bug, typed nil error won't equal to nil.

[the remark this answers is not in the document]
Comment (kelwang) [reply bolt/basic_auth_service.go]: but, maybe you are right. I am not really sure when that happens

[the remark this answers is not in the document]
Comment (desa) [reply bolt/basic_auth_service.go]: Does it need it?

[the remark this answers is not in the document]
Comment (kelwang) [reply bolt/basic_auth_service.go]: ops, i thought i just removed it, let me re-push.

**Pull request merged**